## Fair - VOTE EQUATION MODEL 2016

The equation to predict the 2016 presidential election was
VP = 42.39 + .667*G - .690*P + 0.968*Z

### Input values:

0.97 -  growth rate of real per capita GDP in the first 3 quarters of 2016 (annual rate) (G)

1.42 - growth rate of the GDP deflator in the first 15 quarters of the second Obama administration, 2013:1-2016:3 (annual rate) (P)

2 -  number of quarters in the first 15 quarters of the  administration in which the growth
rate of real per capita GDP is greater than 3.2 percent at an annual rate (Z)

### Variable	Description

G = Growth rate of real per capita GDP in the first three quarters of the on-term election year (annual rate)

P = absolute value of the growth rate of the GDP deflator in the first 15 quarters of the administration (annual rate) except for 1920, 1944, and 1948, where the values are zero.

Z = Number of quarters in the first 15 quarters of the administration in which the growth rate of real per capita GDP is greater than 3.2 percent at an annual rate except for 1920, 1944, and 1948, where the values are zero

I = 1 if there is a Democratic presidential incumbent at the time of the election and -1 if there is a Republican presidential incumbent

DPER = 1 if a Democratic presidential incumbent is running again, -1 if a Republican presidential incumbent is running again, and 0 otherwise

DUR = 0 if either party has been in the White House for one term, 1 [-1] if the Democratic [Republican] party has been in the White House for two consecutive terms, 1.25 [-1.25] if the Democratic [Republican] party has been in the White House for three consecutive terms, 1.50 [-1.50] if the Democratic [Republican] party has been in the White House for four consecutive terms, and so on

WAR	 = 1 for the elections of 1918, 1920, 1942, 1944, 1946, and 1948, and 0 otherwise

VP = Democratic share of the two-party presidential vote

In [41]:
import pandas as pd
import numpy as np

In [44]:
# Economic Indicator Variables
G= 0.97
P= 1.42
Z = 2

In [45]:
#Static Features
I = 1
DPER = 0
DUR = 1
WAR = 0

In [46]:
VP = 42.39 + .667*G + -.690*P +0.968*Z

In [47]:
print('The Democratic share of the two-part presidential vote for 2016 is ' + str(VP)+'%')

The Democratic share of the two-part presidential vote for 2016 is 43.993190000000006%


####  Predicted vote share of the two party presidential vore in 2016 is 43.99%

In [48]:
## Determine Regression Estimates for Fair Vote Equation Model

In [49]:
import statsmodels.api as sm
import matplotlib.pyplot as plt
from statsmodels.sandbox.regression.predstd import wls_prediction_std

In [50]:
fair_df=pd.read_csv("FD2012.csv")
fair_df

t      VP      VC  I  DPER   DUR  WAR       G      P   Z
0   1916  51.682  48.881  1     1  0.00    0   2.229  4.252   3
1   1920  36.148  37.957  1     0  1.00    1 -11.463  0.000   0
2   1924  41.737  42.093 -1    -1  0.00    0  -3.872  5.161  10
3   1928  41.244  42.838 -1     0 -1.00    0   4.623  0.183   7
4   1932  59.149  56.874 -1    -1 -1.25    0 -14.350  6.928   4
5   1936  62.226  58.476  1     1  0.00    0  11.682  2.498   9
6   1940  54.983  52.967  1     1  1.00    0   3.913  0.051   8
7   1944  53.778  51.718  1     1  1.25    1   4.122  0.000   0
8   1948  52.319  53.190  1     1  1.50    1   3.214  0.000   0
9   1952  44.710  49.944  1     0  1.75    0   0.997  2.353   7
10  1956  42.906  50.970 -1    -1  0.00    0  -1.252  1.907   5
11  1960  50.087  54.790 -1     0 -1.00    0   0.674  1.980   5
12  1964  61.203  57.324  1     1  0.00    0   5.030  1.241   9
13  1968  49.425  50.921  1     0  1.00    0   5.045  3.086   7
14  1972  38.209  52.660 -1    -1  0.00    0   5.834  4.813   4
15  1976  51.049  56.850 -1     0 -1.00    0   3.817  7.463   5
16  1980  44.842  51.383  1     1  0.00    0  -3.583  7.795   5
17  1984  40.877  52.778 -1    -1  0.00    0   5.550  5.210   8
18  1988  46.168  54.011 -1     0 -1.00    0   2.403  2.871   5
19  1992  53.621  52.744 -1    -1 -1.25    0   3.035  3.193   3
20  1996  54.737  50.158  1     1  0.00    0   3.315  2.031   4
21  2000  50.262  49.819  1     0  1.00    0   2.031  1.683   7
22  2004  48.767  48.632 -1    -1  0.00    0   2.086  2.141   2
23  2008  53.689  55.535 -1     0 -1.00    0  -1.787  2.745   2
24  2012  52.010  50.681  1     1  0.00    0   1.422  1.470   1

In [51]:
fair_df['GI']=fair_df['G']*fair_df['I']
fair_df['PI']=fair_df['P']*fair_df['I']
fair_df['ZI'] =fair_df['Z']*fair_df['I']

In [52]:
fair_df.head(40)

t      VP      VC  I  DPER   DUR  WAR       G      P   Z      GI  \
0   1916  51.682  48.881  1     1  0.00    0   2.229  4.252   3   2.229   
1   1920  36.148  37.957  1     0  1.00    1 -11.463  0.000   0 -11.463   
2   1924  41.737  42.093 -1    -1  0.00    0  -3.872  5.161  10   3.872   
3   1928  41.244  42.838 -1     0 -1.00    0   4.623  0.183   7  -4.623   
4   1932  59.149  56.874 -1    -1 -1.25    0 -14.350  6.928   4  14.350   
5   1936  62.226  58.476  1     1  0.00    0  11.682  2.498   9  11.682   
6   1940  54.983  52.967  1     1  1.00    0   3.913  0.051   8   3.913   
7   1944  53.778  51.718  1     1  1.25    1   4.122  0.000   0   4.122   
8   1948  52.319  53.190  1     1  1.50    1   3.214  0.000   0   3.214   
9   1952  44.710  49.944  1     0  1.75    0   0.997  2.353   7   0.997   
10  1956  42.906  50.970 -1    -1  0.00    0  -1.252  1.907   5   1.252   
11  1960  50.087  54.790 -1     0 -1.00    0   0.674  1.980   5  -0.674   
12  1964  61.203  57.324  1     1  0.00    0   5.030  1.241   9   5.030   
13  1968  49.425  50.921  1     0  1.00    0   5.045  3.086   7   5.045   
14  1972  38.209  52.660 -1    -1  0.00    0   5.834  4.813   4  -5.834   
15  1976  51.049  56.850 -1     0 -1.00    0   3.817  7.463   5  -3.817   
16  1980  44.842  51.383  1     1  0.00    0  -3.583  7.795   5  -3.583   
17  1984  40.877  52.778 -1    -1  0.00    0   5.550  5.210   8  -5.550   
18  1988  46.168  54.011 -1     0 -1.00    0   2.403  2.871   5  -2.403   
19  1992  53.621  52.744 -1    -1 -1.25    0   3.035  3.193   3  -3.035   
20  1996  54.737  50.158  1     1  0.00    0   3.315  2.031   4   3.315   
21  2000  50.262  49.819  1     0  1.00    0   2.031  1.683   7   2.031   
22  2004  48.767  48.632 -1    -1  0.00    0   2.086  2.141   2  -2.086   
23  2008  53.689  55.535 -1     0 -1.00    0  -1.787  2.745   2   1.787   
24  2012  52.010  50.681  1     1  0.00    0   1.422  1.470   1   1.422   

       PI  ZI  
0   4.252   3  
1   0.000   0  
2  -5.161 -10  
3  -0.183  -7  
4  -6.928  -4  
5   2.498   9  
6   0.051   8  
7   0.000   0  
8   0.000   0  
9   2.353   7  
10 -1.907  -5  
11 -1.980  -5  
12  1.241   9  
13  3.086   7  
14 -4.813  -4  
15 -7.463  -5  
16  7.795   5  
17 -5.210  -8  
18 -2.871  -5  
19 -3.193  -3  
20  2.031   4  
21  1.683   7  
22 -2.141  -2  
23 -2.745  -2  
24  1.470   1

## 2016 Model Estimates based on 1916-2012 data

In [53]:
fairadj2016_df=fair_df[0:25]
fairadj2016_df.head(50)

t      VP      VC  I  DPER   DUR  WAR       G      P   Z      GI  \
0   1916  51.682  48.881  1     1  0.00    0   2.229  4.252   3   2.229   
1   1920  36.148  37.957  1     0  1.00    1 -11.463  0.000   0 -11.463   
2   1924  41.737  42.093 -1    -1  0.00    0  -3.872  5.161  10   3.872   
3   1928  41.244  42.838 -1     0 -1.00    0   4.623  0.183   7  -4.623   
4   1932  59.149  56.874 -1    -1 -1.25    0 -14.350  6.928   4  14.350   
5   1936  62.226  58.476  1     1  0.00    0  11.682  2.498   9  11.682   
6   1940  54.983  52.967  1     1  1.00    0   3.913  0.051   8   3.913   
7   1944  53.778  51.718  1     1  1.25    1   4.122  0.000   0   4.122   
8   1948  52.319  53.190  1     1  1.50    1   3.214  0.000   0   3.214   
9   1952  44.710  49.944  1     0  1.75    0   0.997  2.353   7   0.997   
10  1956  42.906  50.970 -1    -1  0.00    0  -1.252  1.907   5   1.252   
11  1960  50.087  54.790 -1     0 -1.00    0   0.674  1.980   5  -0.674   
12  1964  61.203  57.324  1     1  0.00    0   5.030  1.241   9   5.030   
13  1968  49.425  50.921  1     0  1.00    0   5.045  3.086   7   5.045   
14  1972  38.209  52.660 -1    -1  0.00    0   5.834  4.813   4  -5.834   
15  1976  51.049  56.850 -1     0 -1.00    0   3.817  7.463   5  -3.817   
16  1980  44.842  51.383  1     1  0.00    0  -3.583  7.795   5  -3.583   
17  1984  40.877  52.778 -1    -1  0.00    0   5.550  5.210   8  -5.550   
18  1988  46.168  54.011 -1     0 -1.00    0   2.403  2.871   5  -2.403   
19  1992  53.621  52.744 -1    -1 -1.25    0   3.035  3.193   3  -3.035   
20  1996  54.737  50.158  1     1  0.00    0   3.315  2.031   4   3.315   
21  2000  50.262  49.819  1     0  1.00    0   2.031  1.683   7   2.031   
22  2004  48.767  48.632 -1    -1  0.00    0   2.086  2.141   2  -2.086   
23  2008  53.689  55.535 -1     0 -1.00    0  -1.787  2.745   2   1.787   
24  2012  52.010  50.681  1     1  0.00    0   1.422  1.470   1   1.422   

       PI  ZI  
0   4.252   3  
1   0.000   0  
2  -5.161 -10  
3  -0.183  -7  
4  -6.928  -4  
5   2.498   9  
6   0.051   8  
7   0.000   0  
8   0.000   0  
9   2.353   7  
10 -1.907  -5  
11 -1.980  -5  
12  1.241   9  
13  3.086   7  
14 -4.813  -4  
15 -7.463  -5  
16  7.795   5  
17 -5.210  -8  
18 -2.871  -5  
19 -3.193  -3  
20  2.031   4  
21  1.683   7  
22 -2.141  -2  
23 -2.745  -2  
24  1.470   1

In [54]:
y= fairadj2016_df["VP"]
X = fairadj2016_df[['GI','PI','ZI', 'DPER', 'DUR', 'I', 'WAR']]
X = sm.add_constant(X)

In [55]:
model2016 = sm.OLS(y, X)
results2016 = model2016.fit()
print(results2016.summary())

                            OLS Regression Results                            
Dep. Variable:                     VP   R-squared:                       0.897
Model:                            OLS   Adj. R-squared:                  0.854
Method:                 Least Squares   F-statistic:                     21.07
Date:                Mon, 21 Sep 2020   Prob (F-statistic):           3.30e-07
Time:                        07:58:03   Log-Likelihood:                -54.772
No. Observations:                  25   AIC:                             125.5
Df Residuals:                      17   BIC:                             135.3
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         47.7540      0.603     79.154      0.0

### 2016 Extended Equation
VP = 47.754 + 0.667 (G*I) – 0.690 (P*I) + 0.968(Z*I) + 3.008(DPER) – 3.805 (DUR) -1.563 (I) + 4.8917(WAR)

### Static Features 2016 entered
I = 1
DPER = 0
DUR = 1
WAR = 0

VP = 47.754 + 0.667 (G*I) – 0.690 (P*I) + 0.968(Z*I) - 3.805 – 1.563 

VP = 42.39 + 0.667 (G*I) – 0.690 (P*I) + 0.968(Z*I)


##### Estimates Match
Constant is equal to Intercept + 3.008(DPER) – 3.805 (DUR) -1.563 (I) + 4.8917(WAR)

## Fair - VOTE EQUATION MODEL 2020
The equation to predict the 2020 presidential election is
VP = 45.60 - .673*G + .721*P - 0.792*Z

### Your input values:
G = 100*[{(1-.052)*(1+a/100)*(1+b/100)}**(1/3)-1]

a = -32.9   growth rate of real per capita GDP in the second quarter of 2020 (annual rate) 

b = 25   =   growth rate of real per capita GDP in the third quarter of 2020 (annual rate)

P = 1.7  -    growth rate of the GDP deflator in the first 15 quarters of the Trump administration, 2017:1-2020:3 (annual rate) 

Z = 3 - Number of quarters in the first 15 quarters of the Trump administration in which the growth
rate of real per capita GDP is greater than 3.2 percent at an annual rate 

In [60]:
# Indicator Variables 2020
G= -7.36
P= 1.7
Z = 3

In [61]:
#Static Features 2020
I = -1
DPER = -1
DUR = 0
WAR = 0

## 2020 Model Estimates based on 1916-2016 data

In [62]:
fair_df=pd.read_csv("FD2016.csv")
fair_df

t      VP      VC  I  DPER   DUR  WAR       G      P   Z
0   1916  51.682  48.881  1     1  0.00    0   2.229  4.252   3
1   1920  36.148  37.957  1     0  1.00    1 -11.463  0.000   0
2   1924  41.737  42.093 -1    -1  0.00    0  -3.872  5.161  10
3   1928  41.244  42.838 -1     0 -1.00    0   4.623  0.183   7
4   1932  59.149  56.874 -1    -1 -1.25    0 -14.361  6.926   4
5   1936  62.226  58.476  1     1  0.00    0  11.616  2.467   9
6   1940  54.983  52.967  1     1  1.00    0   3.963  0.041   8
7   1944  53.778  51.718  1     1  1.25    1   4.067  0.000   0
8   1948  52.319  53.190  1     1  1.50    1   3.348  0.000   0
9   1952  44.710  49.944  1     0  1.75    0   1.027  2.369   7
10  1956  42.906  50.970 -1    -1  0.00    0  -1.250  1.894   5
11  1960  50.087  54.790 -1     0 -1.00    0   0.643  1.971   5
12  1964  61.203  57.324  1     1  0.00    0   5.099  1.234   9
13  1968  49.425  50.921  1     0  1.00    0   5.107  3.085   7
14  1972  38.209  52.660 -1    -1  0.00    0   5.862  4.812   4
15  1976  51.049  56.850 -1     0 -1.00    0   3.828  7.476   5
16  1980  44.842  51.383  1     1  0.00    0  -3.596  7.827   5
17  1984  40.877  52.778 -1    -1  0.00    0   5.438  5.277   8
18  1988  46.168  54.011 -1     0 -1.00    0   2.342  2.817   4
19  1992  53.621  52.744 -1    -1 -1.25    0   3.053  3.210   3
20  1996  54.737  50.158  1     1  0.00    0   3.300  2.040   4
21  2000  50.262  49.819  1     0  1.00    0   2.069  1.642   7
22  2004  48.767  48.632 -1    -1  0.00    0   2.118  2.085   2
23  2008  53.689  55.535 -1     0 -1.00    0  -1.701  2.692   2
24  2012  52.010  50.681  1     1  0.00    0   1.094  1.442   2
25  2016  51.163  50.546  1     0  1.00    0   1.208  1.411   2

In [63]:
fair_df['GI']=fair_df['G']*fair_df['I']
fair_df['PI']=fair_df['P']*fair_df['I']
fair_df['ZI'] =fair_df['Z']*fair_df['I']

In [64]:
fair_df.tail()

t      VP      VC  I  DPER  DUR  WAR      G      P  Z     GI     PI  ZI
21  2000  50.262  49.819  1     0  1.0    0  2.069  1.642  7  2.069  1.642   7
22  2004  48.767  48.632 -1    -1  0.0    0  2.118  2.085  2 -2.118 -2.085  -2
23  2008  53.689  55.535 -1     0 -1.0    0 -1.701  2.692  2  1.701 -2.692  -2
24  2012  52.010  50.681  1     1  0.0    0  1.094  1.442  2  1.094  1.442   2
25  2016  51.163  50.546  1     0  1.0    0  1.208  1.411  2  1.208  1.411   2

In [65]:
fairadj2020_df=fair_df[0:26]
fairadj2020_df.head(50)

t      VP      VC  I  DPER   DUR  WAR       G      P   Z      GI  \
0   1916  51.682  48.881  1     1  0.00    0   2.229  4.252   3   2.229   
1   1920  36.148  37.957  1     0  1.00    1 -11.463  0.000   0 -11.463   
2   1924  41.737  42.093 -1    -1  0.00    0  -3.872  5.161  10   3.872   
3   1928  41.244  42.838 -1     0 -1.00    0   4.623  0.183   7  -4.623   
4   1932  59.149  56.874 -1    -1 -1.25    0 -14.361  6.926   4  14.361   
5   1936  62.226  58.476  1     1  0.00    0  11.616  2.467   9  11.616   
6   1940  54.983  52.967  1     1  1.00    0   3.963  0.041   8   3.963   
7   1944  53.778  51.718  1     1  1.25    1   4.067  0.000   0   4.067   
8   1948  52.319  53.190  1     1  1.50    1   3.348  0.000   0   3.348   
9   1952  44.710  49.944  1     0  1.75    0   1.027  2.369   7   1.027   
10  1956  42.906  50.970 -1    -1  0.00    0  -1.250  1.894   5   1.250   
11  1960  50.087  54.790 -1     0 -1.00    0   0.643  1.971   5  -0.643   
12  1964  61.203  57.324  1     1  0.00    0   5.099  1.234   9   5.099   
13  1968  49.425  50.921  1     0  1.00    0   5.107  3.085   7   5.107   
14  1972  38.209  52.660 -1    -1  0.00    0   5.862  4.812   4  -5.862   
15  1976  51.049  56.850 -1     0 -1.00    0   3.828  7.476   5  -3.828   
16  1980  44.842  51.383  1     1  0.00    0  -3.596  7.827   5  -3.596   
17  1984  40.877  52.778 -1    -1  0.00    0   5.438  5.277   8  -5.438   
18  1988  46.168  54.011 -1     0 -1.00    0   2.342  2.817   4  -2.342   
19  1992  53.621  52.744 -1    -1 -1.25    0   3.053  3.210   3  -3.053   
20  1996  54.737  50.158  1     1  0.00    0   3.300  2.040   4   3.300   
21  2000  50.262  49.819  1     0  1.00    0   2.069  1.642   7   2.069   
22  2004  48.767  48.632 -1    -1  0.00    0   2.118  2.085   2  -2.118   
23  2008  53.689  55.535 -1     0 -1.00    0  -1.701  2.692   2   1.701   
24  2012  52.010  50.681  1     1  0.00    0   1.094  1.442   2   1.094   
25  2016  51.163  50.546  1     0  1.00    0   1.208  1.411   2   1.208   

       PI  ZI  
0   4.252   3  
1   0.000   0  
2  -5.161 -10  
3  -0.183  -7  
4  -6.926  -4  
5   2.467   9  
6   0.041   8  
7   0.000   0  
8   0.000   0  
9   2.369   7  
10 -1.894  -5  
11 -1.971  -5  
12  1.234   9  
13  3.085   7  
14 -4.812  -4  
15 -7.476  -5  
16  7.827   5  
17 -5.277  -8  
18 -2.817  -4  
19 -3.210  -3  
20  2.040   4  
21  1.642   7  
22 -2.085  -2  
23 -2.692  -2  
24  1.442   2  
25  1.411   2

In [66]:
y= fairadj2020_df["VP"]
X = fairadj2020_df[['GI','PI','ZI', 'DPER', 'DUR', 'I', 'WAR']]
X = sm.add_constant(X)

In [67]:
model2020 = sm.OLS(y, X)
results2020 = model2020.fit()
print(results2020.summary())

                            OLS Regression Results                            
Dep. Variable:                     VP   R-squared:                       0.862
Model:                            OLS   Adj. R-squared:                  0.808
Method:                 Least Squares   F-statistic:                     16.03
Date:                Mon, 21 Sep 2020   Prob (F-statistic):           1.50e-06
Time:                        08:02:34   Log-Likelihood:                -60.267
No. Observations:                  26   AIC:                             136.5
Df Residuals:                      18   BIC:                             146.6
Df Model:                           7                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         48.0617      0.657     73.132      0.0

### Published Equation
The equation to predict the 2020 presidential election is 
V = 48.06 + 0.673 (G*I) – 0.721 (P*I) + 0.792 (Z*I) + 2.25 (DPER) – 3.76 (DUR) + 0.21 (I) + 3.25 (WAR)

In [81]:
#Static 2020 Variables
I=-1
DPER=-1
DUR=0
WAR=0

In [82]:
#Dynamic Economic Indicators
#Implied value of the growth rate of real per capita GDP in the first 3 quarters of 2020 (annual rate)a
#a = Q2GDP estimate
#b = Q3Gdp estimate

a= -32.9
b = 25

G = 100*((((1-.052)*(1+a/100)*(1+b/100))**(1/3))-1)
print(G)

-7.356783820988744


In [83]:
#Dynamic Economic Indicators

#Implied value of the growth rate of real per capita GDP in the first 3 quarters of 2020 (annual rate) 
G = -7.36

#These are likely non-dynamic
#growth rate of the GDP deflator in the first 15 quarters of the Trump administration
P = 1.7
#number of quarters in the first 15 quarters of the Trump administration in which the growth rate of real per capita GDP is greater than 3.2 percent at an annual rate (Z)
Z = 3

### Fair Model predicted incumbent vote share with updated estimates for 2020

In [88]:
IVS = 48.06 - 0.673*G + 0.721*P - 0.792 *Z + 2.25*DPER - 3.76*DUR + 0.21*I + 3.25*WAR
print ("The incumbent's predicted Vote Share is " + str(IVS))

The incumbent's predicted Vote Share is 49.40298000000001


#### Fair Disclaimer from Dr. Ray Fair
I did not make an economic forecast with my US model this time because the model has nothing to say about the effects of pandemics. I could try to subjectively constant adjust the estimated equations, but this would only be guessing. So I am going to let the user decide what values of G and P to use. The per capita growth rate in the first quarter of 2020 was -5.2 percent at an annual rate. If your guess (annual rates) is -10 percent in the second quarter and -5 percent in the third quarter, then G is (1-.052)(1-.10)(1-.05) raised to the 1/3 power and then subtract 1, which is -6.76 percent. If you use the link "Compute your own prediction," this is done for you, where you need to give as inputs the per capita growth rates at annual rates for the second and third quarters. The value of P through the first quarter of 2020 is 1.9 percent, so using 1.9 will likely be fairly close to the actual. Also note when choosing your estimates of per capita growth rates that population is growing at about 0.5 percent per year at an annual rate. So subtract 0.5 from estimates of non per capita growth rates.